### ISSUE TITLE
calico-node pod keeps restarting

### ISSUE THREAD

**Issue by rkgithubp** [rancher/rke#1327]
**RKE version:**
$ ./rke version
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:30:26Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

**Docker version: (`docker version`,`docker info` preferred)**
$  docker --version
Docker version 18.09.5, build e8ff056

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS EC2 instance

**cluster.yml file:**
nodes:
    - address: ec2-**-**-75-230.compute-1.amazonaws.com
      user: ubuntu
      role:
        - controlplane
        - etcd
    - address: ec2-**-**-43-218.eu-west-1.compute.amazonaws.com
      user: ubuntu
      role:
        - worker
addons: ""
addons_include:
- https://raw.githubusercontent.com/kubernetes/dashboard/master/aio/deploy/recommended/kubernetes-dashboard.yaml

**Steps to Reproduce:**

1. start master node in One AWS region and worker node in another AWS region.
2. install Kubernetes using above cluster.yml
3. check whether installation is ssuccessful.

**Results:**
I am trying to setup Kubernetes cluster on AWS EC2 instances with Master in One region and one Worker in another region.
however the calico-node pod keeps restarting. getting below error:
"Liveness probe failed: Get http://localhost:9099/liveness: dial tcp 127.0.0.1:9099: connect: connection refused"

$ ./rke version
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:30:26Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

$  docker --version
Docker version 18.09.5, build e8ff056

$ kubectl --kubeconfig kube_config_cluster.yml version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:30:26Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl --kubeconfig kube_config_cluster.yml get nodes
NAME                                               STATUS   ROLES               AGE   VERSION
ec2-***.eu-west-1.compute.amazonaws.com   Ready    worker              25m   v1.13.4
ec2-***.compute-1.amazonaws.com           Ready    controlplane,etcd   25m   v1.13.4

$ kubectl --kubeconfig kube_config_cluster.yml get pods -n kube-system
NAME                                        READY   STATUS             RESTARTS   AGE
canal-46f8w                                 2/2     Running            0          25m
canal-lw4w8                                 0/2     CrashLoopBackOff   17         25m
kube-dns-5fd74c7488-qmbq8                   0/3     CrashLoopBackOff   17         25m
kube-dns-autoscaler-c89df977f-zmmkq         1/1     Running            0          25m
kubernetes-dashboard-57df4db6b-96sgb        0/1     CrashLoopBackOff   7          25m
metrics-server-7fbd549b78-9trfl             0/1     CrashLoopBackOff   7          25m
rke-ingress-controller-deploy-job-xp4cx     0/1     Completed          0          25m
rke-kubedns-addon-deploy-job-pvt8l          0/1     Completed          0          25m
rke-metrics-addon-deploy-job-p8cjc          0/1     Completed          0          25m
rke-network-plugin-deploy-job-rfs6r         0/1     Completed          0          25m
rke-user-includes-addons-deploy-job-czt5k   0/1     Completed          0          25m

$ kubectl --kubeconfig kube_config_cluster.yml describe pod canal-lw4w8 -n kube-system
Name:               canal-lw4w8
Namespace:          kube-system
Priority:           0
PriorityClassName:  <none>
Node:               ec2-***.eu-west-1.compute.amazonaws.com/***
Start Time:         Mon, 06 May 2019 06:15:59 +0000
Labels:             controller-revision-hash=cb5cf498f
                    k8s-app=canal
                    pod-template-generation=1
Annotations:        scheduler.alpha.kubernetes.io/critical-pod:
Status:             Running
IP:                 ***
Controlled By:      DaemonSet/canal
Init Containers:
  install-cni:
    Container ID:  docker://283ec8ee1684b84ba49ae688082dc1e9e4686ac6852a5d3b17e7f538dac4906d
    Image:         rancher/calico-cni:v3.4.0
    Image ID:      docker-pullable://rancher/calico-cni@sha256:b3f4045680586133039f1e77f3ebb4bb5e6c60fa85f184f395eee32303462715
    Port:          <none>
    Host Port:     <none>
    Command:
      /install-cni.sh
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Mon, 06 May 2019 06:16:04 +0000
      Finished:     Mon, 06 May 2019 06:16:04 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      CNI_CONF_NAME:         10-canal.conflist
      CNI_NETWORK_CONFIG:    <set to the key 'cni_network_config' of config map 'canal-config'>  Optional: false
      KUBERNETES_NODE_NAME:   (v1:spec.nodeName)
      SLEEP:                 false
    Mounts:
      /host/etc/cni/net.d from cni-net-dir (rw)
      /host/opt/cni/bin from cni-bin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from canal-token-4hkz5 (ro)
Containers:
  calico-node:
    Container ID:   docker://37597b7080c9abe879be5c6f3f4302288a7c234d9e4c8a2f6b26a7d82271401f
    Image:          rancher/calico-node:v3.4.0
    Image ID:       docker-pullable://rancher/calico-node@sha256:2f46157483904649f334da83e6b5e522bf20cbd3ea6ec9f43456673933021d5a
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Mon, 06 May 2019 06:41:58 +0000
    Last State:     Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Mon, 06 May 2019 06:35:39 +0000
      Finished:     Mon, 06 May 2019 06:36:48 +0000
    Ready:          False
    Restart Count:  10
    Requests:
      cpu:      250m
    Liveness:   http-get http://localhost:9099/liveness delay=10s timeout=1s period=10s #success=1 #failure=6
    Readiness:  http-get http://localhost:9099/readiness delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      DATASTORE_TYPE:                     kubernetes
      WAIT_FOR_DATASTORE:                 true
      NODENAME:                            (v1:spec.nodeName)
      CALICO_NETWORKING_BACKEND:          none
      CLUSTER_TYPE:                       k8s,canal
      FELIX_IPTABLESREFRESHINTERVAL:      60
      IP:
      CALICO_IPV4POOL_CIDR:               192.168.0.0/16
      CALICO_DISABLE_FILE_LOGGING:        true
      FELIX_DEFAULTENDPOINTTOHOSTACTION:  ACCEPT
      FELIX_IPV6SUPPORT:                  false
      FELIX_LOGSEVERITYSCREEN:            info
      FELIX_HEALTHENABLED:                true
    Mounts:
      /lib/modules from lib-modules (ro)
      /run/xtables.lock from xtables-lock (rw)
      /var/lib/calico from var-lib-calico (rw)
      /var/run/calico from var-run-calico (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from canal-token-4hkz5 (ro)
  kube-flannel:
    Container ID:  docker://9902d3aafa17c9f9aa8647da4a3db7c622e224b9f60e5897c8f6ec275bf7aed3
    Image:         rancher/coreos-flannel:v0.10.0
    Image ID:      docker-pullable://rancher/coreos-flannel@sha256:57f85f0bb7360f2dd378d74ffc01ac99a4c152ad463455ff429867c6879f4386
    Port:          <none>
    Host Port:     <none>
    Command:
      /opt/bin/flanneld
      --ip-masq
      --kube-subnet-mgr
    State:          Running
      Started:      Mon, 06 May 2019 06:41:58 +0000
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Mon, 06 May 2019 06:36:19 +0000
      Finished:     Mon, 06 May 2019 06:36:49 +0000
    Ready:          True
    Restart Count:  9
    Environment:
      POD_NAME:          canal-lw4w8 (v1:metadata.name)
      POD_NAMESPACE:     kube-system (v1:metadata.namespace)
      FLANNELD_IFACE:    <set to the key 'canal_iface' of config map 'canal-config'>  Optional: false
      FLANNELD_IP_MASQ:  <set to the key 'masquerade' of config map 'canal-config'>   Optional: false
    Mounts:
      /etc/kube-flannel/ from flannel-cfg (rw)
      /run/xtables.lock from xtables-lock (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from canal-token-4hkz5 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  lib-modules:
    Type:          HostPath (bare host directory volume)
    Path:          /lib/modules
    HostPathType:
  var-run-calico:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/calico
    HostPathType:
  var-lib-calico:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/calico
    HostPathType:
  xtables-lock:
    Type:          HostPath (bare host directory volume)
    Path:          /run/xtables.lock
    HostPathType:  FileOrCreate
  flannel-cfg:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      canal-config
    Optional:  false
  cni-bin-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /opt/cni/bin
    HostPathType:
  cni-net-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/cni/net.d
    HostPathType:
  canal-token-4hkz5:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  canal-token-4hkz5
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     :NoSchedule
                 :NoExecute
                 CriticalAddonsOnly
                 node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/network-unavailable:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
                 node.kubernetes.io/unschedulable:NoSchedule
Events:
  Type     Reason     Age                   From                                                       Message
  ----     ------     ----                  ----                                                       -------
  Normal   Scheduled  26m                   default-scheduler                                          Successfully assigned kube-system/canal-lw4w8 to ec2-***.eu-west-1.compute.amazonaws.com
  Normal   Pulling    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  pulling image "rancher/calico-cni:v3.4.0"
  Normal   Pulled     26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Successfully pulled image "rancher/calico-cni:v3.4.0"
  Normal   Created    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Created container
  Normal   Started    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Started container
  Normal   Pulling    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  pulling image "rancher/calico-node:v3.4.0"
  Normal   Pulled     26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Successfully pulled image "rancher/calico-node:v3.4.0"
  Normal   Created    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Created container
  Normal   Started    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Started container
  Normal   Pulling    26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  pulling image "rancher/coreos-flannel:v0.10.0"
  Normal   Pulled     26m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Successfully pulled image "rancher/coreos-flannel:v0.10.0"
  Normal   Created    25m (x2 over 26m)     kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Created container
  Normal   Pulled     25m                   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Container image "rancher/coreos-flannel:v0.10.0" already present on machine
  Normal   Started    25m (x2 over 26m)     kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Started container
  Warning  Unhealthy  25m (x5 over 26m)     kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Readiness probe failed: Get http://localhost:9099/readiness: dial tcp 127.0.0.1:9099: connect: connection refused
  Warning  BackOff    11m (x32 over 19m)    kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Back-off restarting failed container
  Warning  Unhealthy  6m18s (x56 over 25m)  kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Liveness probe failed: Get http://localhost:9099/liveness: dial tcp 127.0.0.1:9099: connect: connection refused
  Warning  BackOff    79s (x106 over 25m)   kubelet, ec2-***.eu-west-1.compute.amazonaws.com  Back-off restarting failed container

--------------------------------

kubectl --kubeconfig kube_config_cluster.yml get pods -n kube-system -o wide
NAME                                        READY   STATUS             RESTARTS   AGE   IP              NODE                                               NOMINATED NODE   READINESS GATES
canal-46f8w                                 2/2     Running            0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>
canal-lw4w8                                 0/2     CrashLoopBackOff   20         31m   172.xx.xx.xx    ec2-***.eu-west-1.compute.amazonaws.com   <none>           <none>
kube-dns-5fd74c7488-qmbq8                   1/3     CrashLoopBackOff   21         31m   10.42.1.3       ec2-***.eu-west-1.compute.amazonaws.com   <none>           <none>
kube-dns-autoscaler-c89df977f-zmmkq         1/1     Running            0          31m   10.42.1.4       ec2-***.eu-west-1.compute.amazonaws.com   <none>           <none>
kubernetes-dashboard-57df4db6b-96sgb        0/1     CrashLoopBackOff   8          31m   10.42.1.6       ec2-***.eu-west-1.compute.amazonaws.com   <none>           <none>
metrics-server-7fbd549b78-9trfl             0/1     CrashLoopBackOff   8          31m   10.42.1.2       ec2-***.eu-west-1.compute.amazonaws.com   <none>           <none>
rke-ingress-controller-deploy-job-xp4cx     0/1     Completed          0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>
rke-kubedns-addon-deploy-job-pvt8l          0/1     Completed          0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>
rke-metrics-addon-deploy-job-p8cjc          0/1     Completed          0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>
rke-network-plugin-deploy-job-rfs6r         0/1     Completed          0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>
rke-user-includes-addons-deploy-job-czt5k   0/1     Completed          0          31m   172.xx.xx.xxx   ec2-***.compute-1.amazonaws.com           <none>           <none>

Note: i was able to setup and run PODs for above scenario previously, but now it is not working. Not sure what is going wrong.

**Comment by guglez** [comment]
I have a similar issue on a local RKE cluster. Were you able to fix your issue?

**Comment by rkgithubp** [comment]
> I have a similar issue on a local RKE cluster. Were you able to fix your issue?

No. I've not been able to fix this.

**Comment by milkcha** [comment]
we have the same problem . our rancher version is v2.2.7.

**Comment by LincolnBruce** [comment]
@milkcha @rkgithubp @guglez rancher/calico-node use hostip:9099/liveness as the healthcheck way. then close the livenss of the calico-node, and see it .

**Comment by Allen-yan** [comment]
we have the same problem . our rancher version is v2.2.4.

**Comment by alex-mnt** [comment]
hi guys,

did anybody solved this issue? because of this the container networking was affected.
Is this something that rancher2.2.9 addresses ?

Thank you

**Comment by guglez** [comment]
> hi guys,
> 
> did anybody solved this issue? because of this the container networking was affected.
> Is this something that rancher2.2.9 addresses ?
> 
> Thank you

Hi @alex-mnt 

Did it happen after upgrade of your cluster?

**Comment by alex-mnt** [comment]
@guglez no, this did not happen after a cluster update.
The only service that was added was Jenkins which spawned dynamically slaves and at one point in time one node was not responsive / heavy build load with jenkins slaves.

From what I noticed this was the main triggered which caused the calico-node to be flagged as unhealthy and caused restarts.  This went in a loop of 7 retries and then the cluster networks was not healthy. Noticeable with the communication between the Jenkins master  <-> slaves. 

Fixed it with specifying the "host: localhost" in the calico-node yml file. This was the missing configuration. 

So far the cluster is back to a healthy state.
